tools/postgres/dbtool.js

```diff
@@ -0,0 +1,447 @@
+    url: url.toString(),
+    urlWithoutDatabase: urlWithoutDatabase.toString(),
+    username: url.username,
+    password: url.password,
+    hostname: url.hostname,
+    port: url.port,
+    database: url.pathname.length > 1 ? url.pathname.substring(1) : "webknossos",
+  };
+})();
+
+const program = new Command();
+
+function safeSpawn(command, args, options = {}) {
+  const prc = spawnSync(command, args, { ...options, encoding: "utf-8" });
+  if (prc.status !== 0) {
+    const error = new Error(`Exit code: ${prc.status}\Stderr:\n${prc.stderr}`);
+    error.status = prc.status;
+    error.stderr = prc.stderr;
+    error.stdout = prc.stdout;
+    throw error;
+  }
+  return prc.stdout;
+}
+
+function safePsqlSpawn(args, options = {}) {
+  try {
+    return safeSpawn("psql", args, options);
+  } catch (err) {
+    throw new Error(`PSQL exit code: ${err.status}\nPSQL stderr:\n${err.stderr}`);
+  }
+}
+
+function callPsql(sql) {
+  return safePsqlSpawn([PG_CONFIG.url, "-tAc", sql]);
+}
+
+function dropDb() {
+  console.log(
+    safePsqlSpawn([PG_CONFIG.urlWithoutDatabase, "-c", `DROP DATABASE ${PG_CONFIG.database}`]),
+  );
+}
+
+function ensureDb() {
+  const doesDbExist = safePsqlSpawn([
+    PG_CONFIG.urlWithoutDatabase,
+    "-tAc",
+    `SELECT 1 FROM pg_database WHERE datname='${PG_CONFIG.database}'`,
+  ]).trim();
+  if (doesDbExist === "1") {
+    console.log("Database already exists");
+  } else {
+    console.log(
+      safePsqlSpawn([
+        PG_CONFIG.urlWithoutDatabase,
+        "-tAc",
+        `CREATE DATABASE ${PG_CONFIG.database};`,
+      ]),
+    );
+  }
+
+  const doesSchemaExist = callPsql(
+    "SELECT schema_name FROM information_schema.schemata WHERE schema_name = 'webknossos';",
+  ).trim();
+  if (doesSchemaExist === "webknossos") {
+    console.log("Schema already exists");
+  } else {
+    refreshSchema();
+  }
+}
+
+function prepareTestDb() {
+  ensureDb();
+  refreshSchema();
+
+  const csvFolder = path.join(__dirname, "..", "..", "test", "db");
+  for (const filename of fs.readdirSync(csvFolder)) {
+    if (filename.endsWith(".csv")) {
+      console.log(`IMPORT ${filename}`);
+      safePsqlSpawn(
+        [
+          PG_CONFIG.url,
+          "-c",
+          `SET session_replication_role = replica; COPY webknossos.${filename.slice(
+            0,
+            -4,
+          )} FROM STDOUT WITH CSV HEADER QUOTE ''''`,
+        ],
+        {
+          input: fs.readFileSync(path.join(csvFolder, filename)),
+        },
+      );
+    }
+  }
+  console.log("✨✨ Done preparing test database");
+}
+
+function refreshSchema() {
+  console.log(safePsqlSpawn([PG_CONFIG.url, "-v", "ON_ERROR_STOP=ON", "-f", schemaPath]));
+}
+
+function dumpSchema(databaseUrl, schemaDir, silent = false) {
+  if (!fs.existsSync(schemaDir) || !fs.statSync(schemaDir).isDirectory) {
+    console.error("Schema directory $schemadir does not exist, aborting!");
+    process.exit(1);
+  }
+
+  for (const fn of fs.readdirSync(schemaDir)) {
```

Comment:
```suggestion
  for (const filename of fs.readdirSync(schemaDir)) {
```
(i read it as function first and was confused. I think it’s filename?)